src/rdkafka_ssl.c

```diff
@@ -1114,6 +1165,32 @@ int rd_kafka_ssl_ctx_init (rd_kafka_t *rk, char *errstr, size_t errstr_size) {
         }
 #endif
 
+#if OPENSSL_VERSION_NUMBER >= 0x10100000
+        if(rk->rk_conf.ssl.openssl_engine_location && !rk->rk_conf.ssl.ptr_engine) {
+                char* engineLoadErrStr = NULL;
+                rk->rk_conf.ssl.ptr_engine = ENGINE_by_id("dynamic");
+                if (!rk->rk_conf.ssl.ptr_engine)
+                        engineLoadErrStr = "ENGINE_by_id failed";
+
+                if (!ENGINE_ctrl_cmd_string(rk->rk_conf.ssl.ptr_engine, "SO_PATH", rk->rk_conf.ssl.openssl_engine_location, 0))
+                        engineLoadErrStr = "ENGINE_ctrl_cmd_string SO_PATH failed";
+
+                if (!ENGINE_ctrl_cmd_string(rk->rk_conf.ssl.ptr_engine, "LIST_ADD", "1", 0))
+                        engineLoadErrStr = "ENGINE_ctrl_cmd_string LIST_ADD failed";
+
+                if (!ENGINE_ctrl_cmd_string(rk->rk_conf.ssl.ptr_engine, "LOAD", NULL, 0))
+                        engineLoadErrStr = "ENGINE_ctrl_cmd_string LOAD failed";
+
```

Comment:
Are the above ctrl_cmd_string()s optional? If not then the code should not continue on error since the error strings will be overwritten.

Reply:
thanks for pointing. they are not optional.
I will make the change to not to continue on error